patches/server/0466-Fix-SpawnChangeEvent-not-firing-for-all-use-cases.patch

```diff
@@ -5,26 +5,32 @@ Subject: [PATCH] Fix SpawnChangeEvent not firing for all use-cases
 
 
 diff --git a/src/main/java/net/minecraft/server/level/ServerLevel.java b/src/main/java/net/minecraft/server/level/ServerLevel.java
-index 5537903d6e1233489d31127f3af3d99c40258eb4..12534be05aa56930cde39e6fef90d402b5eb7c4e 100644
+index 5537903d6e1233489d31127f3af3d99c40258eb4..4d5203493472dde8d80a463027156f356400f269 100644
 --- a/src/main/java/net/minecraft/server/level/ServerLevel.java
 +++ b/src/main/java/net/minecraft/server/level/ServerLevel.java
-@@ -1893,6 +1893,7 @@ public class ServerLevel extends Level implements WorldGenLevel {
+@@ -1890,9 +1890,11 @@ public class ServerLevel extends Level implements WorldGenLevel {
+     public void setDefaultSpawnPos(BlockPos pos, float angle) {
+         // Paper - configurable spawn radius
+         BlockPos prevSpawn = this.getSharedSpawnPos();
++        float prevSpawnAngle = this.getSharedSpawnAngle(); // Paper
```

Comment:
If we cannot completely rely on the prevSpawn anymore, is there any reason to not just use the existing logic in
CraftWorld#getSpawnLocation here ? That location includes the angle (hi doc lol)  (it would invoke ServerLevel#getSharedSpawnPos again, this isn't hot code tho).

Tho tbh, does not really matter, just kinda sad we don't have a MCUtil constructor for locations that includes yaw and pitch. Dunno if that is worth adding ? I am sure it has its usages down the line. 

Purely stylistic feedback tho, functionality looks great :+1: 
I am rather bed bound due to COVID getting me after three years so i don't know if I can test this today, if not I'll try by tomorrow to get this approved and merged :)

Reply:
Oops i wasn't aware that spigot has implemented that so recently, i thought the reverse things. There's no reason i use that instead of the bukkit way (except maybe the previous statement).
We could maybe find more usage for a MCUtil helper but i think the new Position API should clean this a little bit more in general.
Take your time if you need some rest :heart: